src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -184,7 +184,11 @@ public function disableBulkActions()
      */
     public function setDefaultPageLength($value)
     {
-        $this->setOperationSetting('defaultPageLength', $value);
+        if ($value !== 0) {
+            $this->setOperationSetting('defaultPageLength', $value);
+        } else {
+            abort(500, 'You should not use 0 as a key in paginator. If you are looking for "ALL" option, use -1 instead.');
+        }
```

Comment:
I agree as a principle we should try to do this more often - "return early". If we can do an `if` instead of `if else` we should do it. Good catch @promatik 👍